patroni/postgresql.py

```diff
@@ -622,12 +626,12 @@ def restart(self):
 
     def _write_postgresql_conf(self):
         # rename the original configuration if it is necessary
-        if not os.path.exists(self._postgresql_base_conf):
+        if not self._postgresql_custom_conf and not os.path.exists(self._postgresql_base_conf):
```

Comment:
What if there is no `postgresql.conf` in the data directory? I think it is exactly your case...